trio_async/activities.py

```diff
@@ -0,0 +1,51 @@
+import asyncio
+import time
+
+import trio
+import trio_asyncio
+from temporalio import activity
+
+
+# An asyncio-based async activity
+@activity.defn
+async def say_hello_activity_async(name: str) -> str:
+    # Demonstrate a sleep in both asyncio and Trio, showing that both asyncio
+    # and Trio primitives can be used
+
+    # First asyncio
+    activity.logger.info("Sleeping in asyncio")
+    await asyncio.sleep(0.1)
+
+    # Now Trio. We have to invoke the function separately decorated.
+    # Unfortunately the @trio_as_aio decorator doesn't do a proper functools
+    # wrap so it doesn't respond to things like __name__ that @activity.defn
+    # needs.
```

Comment:
Let's not digress into criticizing 3rd party libraries -- they may well have a reason for their implementation/design decisions.

Let's either delete this, or clarify what it is that users cannot do that they might expect to be able to (it's not quite clear to me).

```suggestion
```

Reply:
It's explaining why we can't use the decorator on activities. I can remove the "Unfortunately" bit but it's not criticizing the library, it's an unfortunate situation for those relying on `__name__`.

Reply:
You'd need to remove "proper" also. But our samples repos just need to be teaching people how to use Temporal; they don't need to explain why things do or don't work.

But in any case, I'm familiar with SDK internals and as a reader of this I don't understand what "doesn't respond to things like __name__ that @activity.defn needs" means.

Reply:
I can remove those words, no prob, but I do want to explain to people why they can't use the documented way in trio_asyncio which is to use the decorator

Reply:
OK cool, sounds good. Is the issue that people might expect to stack the decorators and that doesn't work? You can avoid using separate sync and async versions of the helper functions like this:

```python
return await trio_asyncio.trio_as_aio(say_hello_in_trio)(name)
```

Reply:
> Is the issue that people might expect to stack the decorators and that doesn't work?

Yup, and I just want to mention to them why it doesn't. Otherwise, we should stack the decorators here IMO. Yes, what you have also works, I just chose to use the decorator on that other function instead of this. I will remove the couple of offending words.

Reply:
Updated